server/templates/assignment_details_email.html

```diff
@@ -27,12 +27,14 @@
 {%  endif %}
 {% if event and event.location and event.location[0] %}
     <p>
-        <b><a href="https://www.google.com.au/maps/?q={{ event.location[0].name }}%20{{ event.location[0].address.line[0] }}%20{{ event.location[0].address.area }}%20{{ event.location[0].address.locality }}%20{{ event.location[0].address.postal_code }}%20{{ event.location[0].address.country }}">Location:</a></b><br>
-        {{ event.location[0].name }}<br>
-        {{ event.location[0].address.line[0] }}<br>
-        {{ event.location[0].address.area }}<br>
-        {{ event.location[0].address.locality }} {{ event.location[0].address.postal_code }}<br>
-        {{ event.location[0].address.country }}<br>
+        <b><a href="https://www.google.com.au/maps/?q={% if event.location[0].name %}{{ event.location[0].name }}{% endif %}{% if event.location[0].address %}%20{% if event.location[0].address.line[0] %}{{ event.location[0].address.line[0] }}{% endif %}%20{% if event.location[0].address.area %}{{ event.location[0].address.area }}{% endif %}%20{% if event.location[0].address.locality %}{{ event.location[0].address.locality }}{% endif %}%20{% if event.location[0].address.postal_code %}{{ event.location[0].address.postal_code }}{% endif %}%20{% if event.location[0].address.country %}{{ event.location[0].address.country }}{% endif %}{% endif %}">Location:</a></b><br>
```

Comment:
This functionality already exists in `planning.common.format_address`. If you register that function as a jinja filter you can use it here as well.

You could also add a second parameter to `format_address` that will define what separator to use (which will default to ' '), and can then use the same filter in the second half of this location template (i.e. with '<br>' as the separator)